File: ecommerce_client/src/services/orderService.ts
import api from '@/lib/api'
import type { Order } from '@/types/order.types'

export const orderService = {
  // Place order from cart (checkout)
  placeOrder: async (): Promise<Order> => {
    const response = await api.post<Order>('/orders/checkout')
    return response.data
  },

  // Get single order by code
  getOrderByCode: async (code: string): Promise<Order> => {
    const response = await api.get<Order>(`/orders/${code}`)
    return response.data
  },

  // Get customer's orders by email
  getOrdersByCustomerEmail: async (): Promise<Order[]> => {
    const response = await api.get<Order[]>(`/orders/customer`)
    return response.data
  },

  // Get all orders
  getAllOrders: async (): Promise<Order[]> => {
    const response = await api.get<Order[]>('/orders')
    return response.data
  },


}

File: microservice/order-service/src/main/java/com/shop/order_service/order/service/OrderService.java
package com.shop.order_service.order.service;

import com.shop.order_service.cart.model.CartModel;
import com.shop.order_service.order.dto.response.OrderDtoResponse;
import com.shop.order_service.order.type.OrderStatus;

import java.util.List;

public interface OrderService {

    OrderDtoResponse placeOrder(CartModel cartModel);

    OrderDtoResponse getOrderByCode(String code);

    List<OrderDtoResponse> getOrdersByCustomerEmail(String email);

    List<OrderDtoResponse> getAllOrders();

    OrderDtoResponse updateOrderStatus(String code, OrderStatus status);
}
